### PR TITLE
Fixed PugiXML detection on Mac OS X.

### PR DESCRIPTION
**Pull request by mazen-mardini**
This will fix issue #2016 by using pkg-config to find PugiXML on Mac OS X. Brew does supply a .pc file for pugixml so it will be available.

### CODE REVIEW

File: cmake/FindPugiXML.cmake [hunk 1]
```diff
@@ -1,7 +1,22 @@
...
+
+	find_package(PkgConfig REQUIRED)
+	pkg_check_modules(PC_PUGIXML QUIET pugixml)
+	set(PUGIXML_DEFINITIONS ${PC_PUGIXML_CFLAGS_OTHER})
+	
+	find_path(PUGIXML_INCLUDE_DIR pugixml.hpp HINTS ${PC_PUGIXML_INCLUDEDIR} ${PC_PUGIXML_INCLUDE_DIRS})
+	
+	find_library(PUGIXML_LIBRARIES NAMES pugixml HINTS ${PC_PUGIXML_LIBDIR} ${PC_PUGIXML_LIBRARY_DIRS})
+	
+else(APPLE)
```
Comment (ranisalt) [inline cmake/FindPugiXML.cmake]: Use `else()` only.

File: cmake/FindPugiXML.cmake [hunk 2]
```diff
@@ -1,7 +1,22 @@
...
+
+	find_path(PUGIXML_INCLUDE_DIR NAMES pugixml.hpp)
+	find_library(PUGIXML_LIBRARIES NAMES pugixml)
+	
+endif(APPLE)
```
Comment (ranisalt) [inline cmake/FindPugiXML.cmake]: Use `endif()` only.

File: cmake/FindPugiXML.cmake [hunk 3]
```diff
@@ -1,7 +1,22 @@
-find_path(PUGIXML_INCLUDE_DIR NAMES pugixml.hpp)
-find_library(PUGIXML_LIBRARIES NAMES pugixml)
+if(APPLE)
+
```
Comment (ranisalt) [inline cmake/FindPugiXML.cmake]: Remove empty line.

File: cmake/FindPugiXML.cmake [hunk 4]
```diff
@@ -1,7 +1,22 @@
...
+	find_package(PkgConfig REQUIRED)
+	pkg_check_modules(PC_PUGIXML QUIET pugixml)
+	set(PUGIXML_DEFINITIONS ${PC_PUGIXML_CFLAGS_OTHER})
+	
+	find_path(PUGIXML_INCLUDE_DIR pugixml.hpp HINTS ${PC_PUGIXML_INCLUDEDIR} ${PC_PUGIXML_INCLUDE_DIRS})
+	
+	find_library(PUGIXML_LIBRARIES NAMES pugixml HINTS ${PC_PUGIXML_LIBDIR} ${PC_PUGIXML_LIBRARY_DIRS})
+
```
Comment (ranisalt) [inline cmake/FindPugiXML.cmake]: Remove empty line.

File: cmake/FindPugiXML.cmake [hunk 5]
```diff
@@ -1,7 +1,22 @@
-find_path(PUGIXML_INCLUDE_DIR NAMES pugixml.hpp)
-find_library(PUGIXML_LIBRARIES NAMES pugixml)
+if(APPLE)
+
+	find_package(PkgConfig REQUIRED)
+	pkg_check_modules(PC_PUGIXML QUIET pugixml)
+	set(PUGIXML_DEFINITIONS ${PC_PUGIXML_CFLAGS_OTHER})
+	
+	find_path(PUGIXML_INCLUDE_DIR pugixml.hpp HINTS ${PC_PUGIXML_INCLUDEDIR} ${PC_PUGIXML_INCLUDE_DIRS})
+
```
Comment (ranisalt) [inline cmake/FindPugiXML.cmake]: Remove empty line.

File: cmake/FindPugiXML.cmake [hunk 6]
```diff
@@ -1,7 +1,22 @@
...
+	
+endif(APPLE)
 
 include(FindPackageHandleStandardArgs)
-FIND_PACKAGE_HANDLE_STANDARD_ARGS(PugiXML REQUIRED_VARS PUGIXML_INCLUDE_DIR PUGIXML_LIBRARIES)
 
-mark_as_advanced(PUGIXML_INCLUDE_DIR PUGIXML_LIBRARIES)
+find_package_handle_standard_args(PugiXML REQUIRED_VARS PUGIXML_INCLUDE_DIR PUGIXML_LIBRARIES)
+
+mark_as_advanced(PUGIXML_INCLUDE_DIR PUGIXML_LIBRARIES)
```
Comment (ranisalt) [inline cmake/FindPugiXML.cmake]: Missing EOF newline.

File: cmake/FindPugiXML.cmake [hunk 7]
```diff
@@ -1,7 +1,22 @@
...
+	
+	find_path(PUGIXML_INCLUDE_DIR pugixml.hpp HINTS ${PC_PUGIXML_INCLUDEDIR} ${PC_PUGIXML_INCLUDE_DIRS})
+	
+	find_library(PUGIXML_LIBRARIES NAMES pugixml HINTS ${PC_PUGIXML_LIBDIR} ${PC_PUGIXML_LIBRARY_DIRS})
+	
+else(APPLE)
+
+	find_path(PUGIXML_INCLUDE_DIR NAMES pugixml.hpp)
+	find_library(PUGIXML_LIBRARIES NAMES pugixml)
+
```
Comment (ranisalt) [inline cmake/FindPugiXML.cmake]: Remove empty line.

File: cmake/FindPugiXML.cmake [hunk 8]
```diff
@@ -1,7 +1,22 @@
...
+	find_package(PkgConfig REQUIRED)
+	pkg_check_modules(PC_PUGIXML QUIET pugixml)
+	set(PUGIXML_DEFINITIONS ${PC_PUGIXML_CFLAGS_OTHER})
+	
+	find_path(PUGIXML_INCLUDE_DIR pugixml.hpp HINTS ${PC_PUGIXML_INCLUDEDIR} ${PC_PUGIXML_INCLUDE_DIRS})
+	
+	find_library(PUGIXML_LIBRARIES NAMES pugixml HINTS ${PC_PUGIXML_LIBDIR} ${PC_PUGIXML_LIBRARY_DIRS})
+	
+else(APPLE)
+
```
Comment (ranisalt) [inline cmake/FindPugiXML.cmake]: Remove empty line.

File: cmake/FindPugiXML.cmake [hunk 9]
```diff
@@ -1,7 +1,22 @@
-find_path(PUGIXML_INCLUDE_DIR NAMES pugixml.hpp)
-find_library(PUGIXML_LIBRARIES NAMES pugixml)
+if(APPLE)
+
+	find_package(PkgConfig REQUIRED)
+	pkg_check_modules(PC_PUGIXML QUIET pugixml)
+	set(PUGIXML_DEFINITIONS ${PC_PUGIXML_CFLAGS_OTHER})
```
Comment (ranisalt) [inline cmake/FindPugiXML.cmake]: Try using `add_compile_options(${PC_PUGIXML_CFLAGS_OTHER})` to apply flags directly.

**Comment by ranisalt** [comment]
You have mostly style misses here

**Comment by ranisalt** [comment]
Good to go. I'll wait another Mac OS X user approve it first before merging to be sure.

@marksamman @jakon89

**Comment by ranisalt** [comment]
Thanks!

@mazen-mardini if you have time, can you have a look on using Travis w/ OS X? I feel it would be a good addition to have CI with OS X too 😄

**Comment by mazen-mardini** [comment]
@ranisalt I'll look into it. :)

**Comment by mazen-mardini** [comment]
@ranisalt I've tried to setup Travis CI to build on osx but it simply wont do it. There are 8 jobs, 4 of them (all linux) completed successfully. The other 4 were osx and stayed queued indefinitely. I believe this might be the reason: https://www.traviscistatus.com/incidents/426tczyzd1wq

If you wish to see how I set it up, check out: https://github.com/mazen-mardini/forgottenserver/blob/add_travis_osx_support/.travis.yml

For results: https://travis-ci.org/mazen-mardini/forgottenserver

**Comment by ranisalt** [comment]
The updated results now show it didn't find pugixml, so... maybe you forgot to pull from master?

**Comment by mazen-mardini** [comment]
@ranisalt My bad, I didn't merge those changes to my master-branch. I've now merged all my branches into master and I've also added pkg-config as a dependency on osx. So let us see if it works this time: https://travis-ci.org/mazen-mardini/forgottenserver/builds/200340744

If this works, you might also have to modify the wiki to include pkg-config as a dependency on Mac OS X.

**Comment by ranisalt** [comment]
Sure do, but isn't it installed by default? At least on Linux it's a very basic package for anything related to compiling.

**Comment by mazen-mardini** [comment]
@ranisalt Apparently you're right, I double-checked again and indeed it was installed by default on my system as well as on Travis CI. Okay, so pkg-config wasn't the problem. If I find a fix I'll make a pull request with my .travis.yml.

**Comment by ranisalt** [comment]
I suggest you cleanup your fork, work on a new branch and apply the changes.

**Comment by mazen-mardini** [comment]
@ranisalt You are correct, branching is good practice. But experimenting with the Travis CI build-process was inevitably going to make the repository dirty. If I fix the bug I'll make sure you get a proper pull request, so don't worry about it. :)

**Pull request merged**